osu.Game.Rulesets.Catch/UI/CatcherArea.cs

```diff
@@ -39,17 +41,35 @@ public CatcherArea(BeatmapDifficulty difficulty = null)
             };
         }
 
-        public void Add(DrawableHitObject fruit, Vector2 absolutePosition)
+        public void OnJudgement(DrawableCatchHitObject fruit, Judgement judgement)
         {
-            fruit.RelativePositionAxes = Axes.None;
-            fruit.Position = new Vector2(MovableCatcher.ToLocalSpace(absolutePosition).X - MovableCatcher.DrawSize.X / 2, 0);
+            if (judgement.IsHit)
+            {
+                var screenSpacePosition = fruit.ScreenSpaceDrawQuad.Centre;
+
+                // remove the fruit from its current parent.
+                // todo: make this less ugly, somehow.
+                (fruit.Parent as Container<DrawableHitObject>)?.Remove(fruit);
+                (fruit.Parent as Container)?.Remove(fruit);

```

Comment:
If possible, it would be nice to keep this in CatchPlayfield.OnJudgement in preparation for my scrolling changes, where this will be able to be converted into `HitObjects.Remove(fruit)`.